### PR TITLE
8311074: RISC-V: Fix -Wconversion warnings in some code header files

### PR DESCRIPTION
**Pull request by zifeihan**
Hi, refer to the modification of aarch64 in https://github.com/openjdk/jdk/pull/14675 , here is the riscv version, before this PR has the same -Wconversion warnings as aarch64, after this this PR the -Wconversion warnings has been eliminated.
Please take a look and have some reviews. Thanks a lot.

## Testing:
- [x] Tier1 tests (release)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8311074](https://bugs.openjdk.org/browse/JDK-8311074): RISC-V: Fix -Wconversion warnings in some code header files (**Enhancement** - P4)

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**) ⚠️ Review applies to [d7c47a0d](https://git.openjdk.org/jdk/pull/14702/files/d7c47a0dd70e77e0abc79f4d1caa09093266aef5)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14702/head:pull/14702` \
`$ git checkout pull/14702`

Update a local copy of the PR: \
`$ git checkout pull/14702` \
`$ git pull https://git.openjdk.org/jdk.git pull/14702/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14702`

View PR using the GUI difftool: \
`$ git pr show -t 14702`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14702.diff">https://git.openjdk.org/jdk/pull/14702.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14702#issuecomment-1612292599)

### CODE REVIEW

File: src/hotspot/cpu/riscv/stackChunkFrameStream_riscv.inline.hpp [hunk 1]
```diff
@@ -114,8 +114,8 @@ inline int StackChunkFrameStream<frame_kind>::interpreter_frame_num_oops() const
   f.interpreted_frame_oop_map(&mask);
   return mask.num_oops()
         + 1 // for the mirror oop
-        + ((intptr_t*)f.interpreter_frame_monitor_begin()
-            - (intptr_t*)f.interpreter_frame_monitor_end()) / BasicObjectLock::size();
+        + pointer_delta_as_int((intptr_t*)f.interpreter_frame_monitor_begin(),
+              (intptr_t*)f.interpreter_frame_monitor_end())/BasicObjectLock::size();
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/stackChunkFrameStream_riscv.inline.hpp:118]: Nit: I think you should keep the surrounding spaces for `/`

**Review by RealFYang — APPROVED** [review]

**Review by coleenp — APPROVED** [review]
Looks good.  Sorry I guess I could have done these also.  Thank you.

**Comment by zifeihan** [comment]
@RealFYang @coleenp Thanks for the review.
/integrate

**Pull request closed without merging**